nuxeo-core/nuxeo-core/src/main/java/org/nuxeo/ecm/core/trash/PropertyTrashService.java

```diff
@@ -90,13 +99,13 @@ protected void doTrashDocument(DocumentModel doc) {
     public Set<DocumentRef> undeleteDocuments(List<DocumentModel> docs) {
         Set<DocumentRef> docRefs = new HashSet<>();
         for (DocumentModel doc : docs) {
-            docRefs.addAll(doUntrashDocument(doc));
+            docRefs.addAll(doUntrashDocument(doc, true));
         }
         docs.stream().map(DocumentModel::getCoreSession).findFirst().ifPresent(CoreSession::save);
         return docRefs;
     }
 
-    protected Set<DocumentRef> doUntrashDocument(DocumentModel doc) {
+    protected Set<DocumentRef> doUntrashDocument(DocumentModel doc, boolean processChildren) {
```

Comment:
This isn't getting backported so I suppose its fine to change the method signature.  I might have changed the existing method to call this method with default value for `processChildren`.  That way, existing code the uses it wouldn't break  (it is a `protected` method though, so unlikely to be used elsewhere).

Reply:
Agree, but that's not backported and 10.1 protected method, I'd rather do this than duplicate a method for nothing